osu.Game/Screens/Edit/EditorBeatmap.cs

```diff
@@ -105,7 +105,7 @@ public EditorBeatmap(IBeatmap playableBeatmap, ISkin beatmapSkin = null, Beatmap
                 BeatmapSkin.BeatmapSkinChanged += SaveState;
             }
 
-            beatmapProcessor = playableBeatmap.BeatmapInfo.Ruleset.CreateInstance().CreateBeatmapProcessor(PlayableBeatmap);
+            beatmapProcessor = playableBeatmap.BeatmapInfo.Ruleset.CreateInstance().CreateBeatmapProcessor(this);
```

Comment:
This is the primary fix and it works by addressing the thing mentioned in https://github.com/ppy/osu/issues/20675#issuecomment-1279742039, namely reading from the wrong `BeatmapInfo` instance.

I audited both beatmap processors of ours that actually do stuff (osu! and catch), and all of the properties they touch aside from `BeatmapInfo` are proxied in `EditorBeatmap` forward to `PlayableBeatmap`, so from that angle this seems *safe-ish*.